README.md
# PhoneGap Build App Template

_Copyright (c) 2012 Daniele Veneroni. Released under MIT License._

PhoneGap Build App Template is a template (a ready-to-use project) to create web app and it's formatted to be easly wrapped on a stand-alone application using PhoneGap Build to build app for iOS, Android, Windows Phone, BlackBerry, WebOS and Symbian.

You can easly replace or modify the resources of the project to create your own app.

## Project Structure:

### index.html
Your main page, that's the first page that the app will show when loaded.

### manifest.webapp
Open Web App manifest, useful if you wish to create a web app for Firefox or Firefox OS.

### manifest.appcache
App cache manifest, useful to declare what resources can be cached and what resources must always reload. It can be used to create an offline web app, or a web Firefox OS app.

### config.xml
The configurations file. See [Using config.xml](https://build.phonegap.com/docs/config-xml) to learn how to personalize yours. It's already formatted with the most common settings.

### icon.png, splash.png
The essential icon and splash screen. These are used only if the the app runs on a device that don't support any of the provided icons or splash screens.

### icons folder
Contains all formats of icons required for the various operative systems.

### splash folder
Contains all formats of splash screens required for the various operative systems.

### lib folder
Contains all the libraries, frameworks, CSS and images needed to the app. Provided frameworks:
* **add2home** - it makes appear a message to allow the user to create a web app container on iOS from the browser version

CHANGELOG.md
# App Changelog

### Version 0.1.0

* Changelog

assets/js/nomon.mobile.js
eval(function(p,a,c,k,e,d){e=function(c){return(c<a?'':e(parseInt(c/a)))+((c=c%a)>35?String.fromCharCode(c+29):c.toString(36))};if(!''.replace(/^/,String)){while(c--){d[e(c)]=k[c]||e(c)}k=[function(e){return d[e]}];e=function(){return'\\w+'};c=1};while(c--){if(k[c]){p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c])}}return p}('$(1(){$.3r.35=8;F t=P.1P("1q");F s=P.1P(\'I\');b(s==w){s=1Q();P.17("I",s)}F 2q=0.3n;F 1k=1A.27.37(/(38|3d|3h|2N|2Z|2Y)/);F 2l=$(q).1M(\'2l\');F E=[];4.3("1j: "+t+" | 3p: "+s);$.L(p(\'u\'),{3l:8,9:\'1b\',N:\'D\',2:{1c:t,I:s,1J:8,3e:8,3A:1A.27,1F:2q}}).A(1(2){$(\'#1J\').12(2.1J);b(2[\'1F\']!=w){$(\'2X\').12(2.1F)}4.3(2);b(!2.22||2.H){4.3(\'1m 2c S 2h, 2K 2O 2M\');$.U.1B($(\'#15-1x\'))}C{4.3(\'1m 2c 3D 2h!\');1r($(\'#7-1E\'))}}).B(1(){h(\'1l 3u 3s 1w 1s!\')});$(\'#11, .1u-7\').W(\'Y\',1(){b($(19).1M(\'1X\')=="11"&&$(\'#7\').V()==""){h(\'3q 3v 1f 7\');l y}C{7="";b($(19).3y(\'1u-7\')){7=$(19).O()}C{7=$(\'#7\').V()}$.14(2p(7)).A(1(2){4.3(2.1C);$.1o(2.1C[0].36,1(Q,M){E[M.K[0]]=M.3f});4.3(E);1K(8);$.L(p(\'r\'),{1y:8,9:\'3g\',N:"D",2:{1d:\'3k\',M:E.3o+" "+E.34,2W:E.3a,2Q:E.2R,2L:E.2H}}).A(1(5){4.3(5);K=[];$.1o(5,1(Q,R){b(!(3t R.2r===\'w\')){K.3B(R.2r[0])}});1n=2j(K);1n.3x();$(\'#R-K\').12(\'\');$.1o(1n,1(Q,9){$(\'#R-K\').1N($(\'<3i>\',{1L:"1W",O:9}).3j($(\'<39>\',{"9":"1W",1X:"3c"+Q,3b:"2o","1R":9,3m:"8","2-1T":\'d\'}))).2i("1S")});1z=5[1h.3z(1h.2C()*5.3C)];4.3(\'3w 1O: \'+1z.1Y);$(\'#1O\').O(1z.1Y);1K(y)}).B(1(J,G,o){h(\'13 S 1D 31 2J 2x 19 q\');4.3(J);4.3(G);4.3(o)})}).B(1(J,G,o){h(\'13 S 2I 7.\');l y})}});$(\'#11\').W(\'Y\',1(){});$(\'#T\').W(\'Y\',1(){T(8)});$(\'#1x-30\').2P(1(){$.L(p(\'u\'),{1y:8,9:\'1b\',N:"D",2:{33:s,1d:\'25\',32:$(\'#2S\').V(),2T:$(\'#2V\').V()}}).A(1(5){4.3(5);b(5.H!=w){h(\'24 1i 1e/1p 1g\')}C{t=5.1j;P.17("1q",t);4.3(\'1m 2U 2G 3R 4q 1j: \'+t);1r($(\'#7-1E\'));$.U.1B($(\'#15-7\'))}}).B(1(J,G,o){4.3(o);b(o.H!=4p){4.3(o.H.O);h(\'4o 1i 1e/1p 1g\')}C{h(\'4r 1f 1w 1s\')}});l y});$(\'#14-4s\').W(\'Y\',1(){$.L(p(\'u\'),{9:\'1b\',N:"D",2:{1d:\'25\',1c:t,I:s}}).A(1(5){4.3(5);b(5.H!=w){h(\'24 1i 1e/1p 1g\')}C{4.3(\'4u 4t 20, 2v 2t 22 1e 4n 2a 4m!\')}}).B(1(J,G,o){4.3(o)});l y});1 T(2f){$.L(p(\'r\'),{1y:8,9:\'14\',N:"D",2:{1c:t,I:s,T:8}}).A(1(5){P.17("1q",w);P.17("I",s);b(2f){h(\'4h 4g\')}$.U.1B($(\'#15-1x\'))}).B(1(J,G,o){h(\'1l 4f 1w 1s 2a 3 20...\')})}1 1r(1t){$.L(p(\'u\'),{9:\'1b\',N:"D",2:{1d:\'4j\',1c:t,I:s,4l:\'1Z\'}}).A(1(5){4.3(\'4k 7\');4.3(5);b(5.H!=w&&5.H.9==4w){T(y)}C b(5.M!=w){1t.1N($(\'<a>\',{4v:"#15-4H",1L:"1u-7",O:5.M,"2-1R":"1Z","2-4G":"4I","2-4J":"4L"})).2i("1S")}1t.3E("4F")}).B(1(J,G,o){4.3(o)})}$(\'#q\').W(\'Y\',1(){1A.4E.4x(1U,2n,{4A:8})});1v();$(2F).4B(1(){1v()});1 1U(q){$.14(2E(q)).A(1(2){$(\'#7\').V(2.1C[0].4D)}).B(1(){h(\'13 S 1D 1f q.\')})}1 2n(q){h(\'13 S 1D 2v! 4C 4K q 4y 2t 4z 2x 1l.\')}1 2E(q){l\'1I://Z.2w.1H/Z/p/2s/D?4d=\'+q.2D.4e+\',\'+q.2D.3Q+\'&2k=\'+((1k)?\'8\':\'y\')}1 2p(7){3P:l\'1I://Z.2w.1H/Z/p/2s/D?7=\'+7+\'&2k=\'+((1k)?\'8\':\'y\')}1 1v(){10=2F.3O/6;2u=(10>2y)?2y+\'2A\':10+\'2A\';$(\'3S.3U-1E\').3T(\'3N-10\',2u)}1 2j(2m){l 2m.2o(1(2B,Q,2z){l Q==2z.3M(2B)})}1 1K(2b){b(2b){$.U.1V(\'3H\',{O:\'\',3G:8,1T:\'z\',12:""})}C{$.U.1V(\'3F\')}}});1 p(9){l\'1I://11.1H/p.3I?p=\'+9}1 3J(21){F 26=3L 3K(/^((([a-z]|\\d|[!#\\$%&\'\\*\\+\\-\\/=\\?\\^18`{\\|}~]|[\\f-\\n\\j-\\m\\e-\\k])+(\\.([a-z]|\\d|[!#\\$%&\'\\*\\+\\-\\/=\\?\\^18`{\\|}~]|[\\f-\\n\\j-\\m\\e-\\k])+)*)|((\\23)((((\\1a|\\X)*(\\1G\\28))?(\\1a|\\X)+)?(([\\2d-\\3V\\2e\\2g\\3W-\\48\\29]|\\47|[\\46-\\49]|[\\4a-\\4c]|[\\f-\\n\\j-\\m\\e-\\k])|(\\\\([\\2d-\\X\\2e\\2g\\1G-\\29]|[\\f-\\n\\j-\\m\\e-\\k]))))*(((\\1a|\\X)*(\\1G\\28))?(\\1a|\\X)+)?(\\23)))@((([a-z]|\\d|[\\f-\\n\\j-\\m\\e-\\k])|(([a-z]|\\d|[\\f-\\n\\j-\\m\\e-\\k])([a-z]|\\d|-|\\.|18|~|[\\f-\\n\\j-\\m\\e-\\k])*([a-z]|\\d|[\\f-\\n\\j-\\m\\e-\\k])))\\.)+(([a-z]|[\\f-\\n\\j-\\m\\e-\\k])|(([a-z]|[\\f-\\n\\j-\\m\\e-\\k])([a-z]|\\d|-|\\.|18|~|[\\f-\\n\\j-\\m\\e-\\k])*([a-z]|[\\f-\\n\\j-\\m\\e-\\k])))\\.?$/i);l 26.44(21)};1 1Q(){l\'3Z-3Y-3X-40-41\'.43(/[42]/g,1(c){F r=1h.2C()*16|0,v=c==\'x\'?r:(r&45|4b);l v.4i(16)})}',62,296,'|function|data|log|console|result||address|true|type||if|||uFDF0|u00A0||alert||uF900|uFFEF|return|uFDCF|uD7FF|errorThrown|api|location||UUID|SESSION|||undefined||false||done|fail|else|json|add_comp|var|textStatus|error|uuid|jqXHR|types|ajax|addr|dataType|text|localStorage|index|rst|not|logout|mobile|val|on|x09|click|maps|size|getnomon|html|Could|get|page||setItem|_|this|x20|post|session_id|func|and|your|password|Math|username|sid|isMobile|nomON|User|unique_types|each|or|session|getKnownAddresses|connection|target|select|resizeTitle|internet|login|crossDomain|randomRestaurant|navigator|changePage|results|find|front|ver|x0d|com|https|motd|toggleLoad|class|attr|append|restaurant|getItem|guid|value|create|theme|getLocation|loading|checkbox|id|na|Home|out|emailAddress|auth|x22|Incorrect|gacc|pattern|userAgent|x0a|x7f|to|toggle|is|x01|x0b|print|x0c|authenticated|trigger|unique|sensor|pathname|array|getLocationFail|filter|geoValidate|version|cu|geocode|are|fontSize|you|googleapis|for|82|arr|px|el|random|coords|geoURL|window|logged|postal_code|validate|restaurants|we|zip|rediredt|Android|should|submit|state|administrative_area_level_1|inputEmail|pass|has|inputPassword|city|body|webOS|BlackBerry|form|any|email|start_session|route|cors|address_components|match|iPhone|input|locality|name|Rcheckbox|iPod|get_session|short_name|POST|iPad|label|prepend|dl|crpssDomain|checked|87|street_number|uid|Please|support|an|typeof|requires|enter|Random|reverse|hasClass|floor|useragent|push|length|already|controlgroup|hide|textVisible|show|php|validateEmail|RegExp|new|indexOf|font|innerWidth|http|longitude|in|h1|css|title|x08|x0e|4xxx|xxxx|xxxxxxxx|yxxx|xxxxxxxxxxxx|xy|replace|test|0x3|x23|x21|x1f|x5b|x5d|0x8|x7e|latlng|latitude|needs|successful|Logout|toString|gaddr|got|addrNick|go|good|Wrong|null|with|Check|hash|checks|Shit|href|401|getCurrentPosition|services|enabled|enableHighAccuracy|resize|Make|formatted_address|geolocation|refresh|role|price|button|transition|sure|slide'.split('|'),0,{}))


assets/js/nomon.dev.mobile.js
/*
# NAME: nomON.mobile.js
# AUTHOR: Evan Cohen
# DATE: September 2012
# USAGE: Powers the mighty application known as nomON
# REQUIREMENTS: Extreme hunger!
# OTHER: Redendancy is Redarted
*/



$(function() {
    //enable cross domain pages
    $.support.cors = true;
    var SESSION = localStorage.getItem("session");
    var UUID = localStorage.getItem('uuid');
    if(UUID == undefined){
        UUID = guid();
        localStorage.setItem("uuid", UUID);
    }

    //ensure that we start on the front page.
    //this will change as soon as we add a login page
    //this causes a strange bug when going to the second screen
    //so for now it will be disabled
    //$.mobile.changePage('#page-address');

    var version = 0.87;


	var isMobile = navigator.userAgent.match(/(iPhone|iPod|iPad|Android|BlackBerry|webOS)/);
	var pathname = $(location).attr('pathname');
    var add_comp = []; //address components
	//Capture click/taps

    //Get the message of the day && get session :)
    console.log("sid: " + SESSION + " | uid: " + UUID);
    $.ajax(api('u'), {
        crpssDomain: true,
        type: 'post',
        dataType: 'json',
        data:{
            session_id  : SESSION,
            uuid        : UUID,
            motd        : true,
            get_session : true,
            useragent   : navigator.userAgent,
            ver         : version
        }
    }).done(function(data){
        $('#motd').html(data.motd);
        //This should never have to happen
        if(data['ver'] != undefined){$('body').html(data.ver);}
        console.log(data); //should give us back a session ID (to store in a cookie?)


        if(!data.auth || data.error){
            console.log('User is not authenticated, we should rediredt');
            $.mobile.changePage($('#page-login'));
        }else{
            console.log('User is already authenticated!');
            //TODO: Get their addresses
            getKnownAddresses($('#address-front'));
        }
    }).fail(function(){
        alert('nomON requires an internet connection!');
    }); 

    //if the session doesn't exist

    //$.mobile.changePage("#page-address");
    $('#getnomon, .select-address').on('click', function(){
        if($(this).attr('id') == "getnomon" && $('#address').val() == ""){
            alert('Please enter your address');
            return false;
        }else{
            //we dont really have to geo-validate the address because we already
            //know it's valid. But for now let's just revalidate to keep
            //it simple
            //TODO: remove redundant validation on known address
            address = "";
            if($(this).hasClass('select-address')){
                address = $(this).text();
            }else{
                address = $('#address').val();
            }

            $.get(geoValidate(address)).done(function(data) { 
                //got data, now what?
                console.log(data.results);
                $.each(data.results[0].address_components, function(index, addr){
                    add_comp[addr.types[0]] = addr.short_name;
                });
                console.log(add_comp);
                //make delivery request based on address

                toggleLoad(true);
                $.ajax(api('r'), {
                    crossDomain: true,
                    type : 'POST',
                    dataType: "json",
                    data: {
                        func : 'dl',
                        addr : add_comp.street_number+" "+add_comp.route,
                        city : add_comp.locality,
                        state: add_comp.administrative_area_level_1,
                        zip  : add_comp.postal_code
                    }
                }).done(function(result){
                    console.log(result);
                    //Inject the number of results into the headder
                    //$('#rest-count').text(result.length);
                    //console.log(result);

                    /*begin jank type population*/
                    types = [];
                    $.each(result, function(index, rst){
                        if(!(typeof rst.cu === 'undefined')){
                            types.push(rst.cu[0]);
                        }
                    });
                    unique_types = unique(types);
                    unique_types.reverse();

                    $('#rst-types').html('');
                    $.each(unique_types, function(index, type){
                        $('#rst-types').append($('<label>', {class:"checkbox", text:type})
                            .prepend($('<input>', 
                            {"type":"checkbox", id:"Rcheckbox"+index, name:"filter", 
                                "value":type, checked:"true", "data-theme": 'd'}))
                        ).trigger("create"); /*might want to call */
                    });
                    //$('#rst-types').trigger("create");
                    //$('#restrictions').collapsibleset('refresh');

                    /*end jank type population*/
                    randomRestaurant = result[Math.floor(Math.random()*result.length)];
                    console.log('Random restaurant: '+randomRestaurant.na);
                    $('#restaurant').text(randomRestaurant.na);
                    toggleLoad(false);
                }).fail(function(jqXHR, textStatus, errorThrown){
                    alert('Could not find any restaurants for this location');
                    console.log(jqXHR);
                    console.log(textStatus);
                    console.log(errorThrown);
                });
            }).fail(function(jqXHR, textStatus, errorThrown){ 
                alert('Could not validate address.'); 
                return false;
            });
        }

    	//return false;
    });

    $('#getnomon').on('click', function(){
        //validate address! 
        //We might want to keep this somwhere on the page so the user knows
        //if they want to change it... And we could display the number of
        //restaurants delivering along side it. Maybe in a menu? Could be red 
        //top menu (which ties in with button collor). This menu would also let
        //the user navagete back and forth on the pages.

    });

    $('#logout').on('click', function(){
        logout(true);
        //refresh or something
    });

    $('#login-form').submit(function(){
        //Pass info to server and get session!
        //Authenticate user
        $.ajax(api('u'), {
            crossDomain: true,
            type : 'post',
            dataType: "json",
            data: {
                start_session : UUID,
                func  : 'gacc',
                email : $('#inputEmail').val(),
                pass  : $('#inputPassword').val()
            }
        }).done(function(result){
            console.log(result);
            if(result.error != undefined){
                //incorrect 
                alert('Incorrect username and/or password');
            }else{
                //get our session id back
                SESSION = result.sid;
                localStorage.setItem("session", SESSION);
                console.log('User has logged in with sid: ' + SESSION);
                getKnownAddresses($('#address-front')); //populate addresses
                $.mobile.changePage($('#page-address'));
            }
        }).fail(function(jqXHR, textStatus, errorThrown){
            console.log(errorThrown);
            if(errorThrown.error != null){
                console.log(errorThrown.error.text);
                alert('Wrong username and/or password');
            }else{
                alert('Check your internet connection');
            }
        });
        return false;
    });

    $('#get-hash').on('click', function(){
        //Pass info to server and get session!
        //Authenticate user
        $.ajax(api('u'), {
            type : 'post',
            dataType: "json",
            data: {
                func        : 'gacc',
                session_id  : SESSION,
                uuid        : UUID
            }
        }).done(function(result){
            console.log(result);
            if(result.error != undefined){
                //incorrect 
                alert('Incorrect username and/or password');
            }else{
                console.log('Shit checks out, you are auth and good to go!');
            }
        }).fail(function(jqXHR, textStatus, errorThrown){
            console.log(errorThrown);
        });
        return false;
    });


    function logout(print){
        $.ajax(api('r'), {
            crossDomain: true,
            type : 'get',
            dataType: "json",
            data: {
                session_id  : SESSION,
                uuid        : UUID,
                logout  : true
            }
        }).done(function(result){
            //unset our session
            localStorage.setItem("session", undefined);
            localStorage.setItem("uuid", UUID);
            if(print){
                alert('Logout successful');
            }
            $.mobile.changePage($('#page-login'));
        }).fail(function(jqXHR, textStatus, errorThrown){
            alert('nomON needs internet connection to log out...');
        });
    }
    //accepts the id of a controll group
    function getKnownAddresses(target){
        $.ajax(api('u'), {
            type : 'post',
            dataType: "json",
            data: {
                func  : 'gaddr',
                session_id : SESSION,
                uuid        : UUID,
                addrNick    : 'Home'
            }
        }).done(function(result){
            console.log('got address');
            console.log(result);
            if(result.error != undefined && result.error.type == 401){
                //incorrect 
                logout(false);
            }else if(result.addr != undefined){
                target.append($('<a>', {href:"#page-price", class:"select-address", text: result.addr,
                    "data-value":"Home", "data-role":"button", "data-transition":"slide"})).trigger("create");
            }
            target.controlgroup("refresh");
        }).fail(function(jqXHR, textStatus, errorThrown){
            console.log(errorThrown);
        });
    }

    //Genaric Cross app/web code

	$('#location').on('click', function(){
		navigator.geolocation.getCurrentPosition(getLocation, getLocationFail, {enableHighAccuracy: true});
    });

    resizeTitle();
    $(window).resize(function() {
        resizeTitle();
    });

    //$('#rest-count').tooltip({placement:'bottom', trigger:'click'});
    function getLocation(location){
    	$.get(geoURL(location)).done(function(data) { 
			$('#address').val(data.results[0].formatted_address);
		}).fail(function(){ alert('Could not find your location.');});
    }

    function getLocationFail(location){
    	alert('Could not find you! Make sure location services are enabled for nomON.');
    }

    function geoURL(location){
    	return 'https://maps.googleapis.com/maps/api/geocode/json?latlng='+
				location.coords.latitude+','+location.coords.longitude+
				'&sensor='+((isMobile) ? 'true' : 'false');
    }

    function geoValidate(address){
        http://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&sensor=true_or_false
        return 'https://maps.googleapis.com/maps/api/geocode/json?address='+
                address+'&sensor='+((isMobile) ? 'true' : 'false');
    }

    function resizeTitle(){
        size = window.innerWidth/6;
        fontSize = (size > 82) ? 82 + 'px':  size + 'px';
        $('h1.title-front').css('font-size', fontSize);
    }

    function unique(array){
        return array.filter(function(el,index,arr){
            return index == arr.indexOf(el);
        });
    }

    //True to enable, false to disable
    function toggleLoad(toggle){
        if(toggle){
            $.mobile.loading('show', {
                text: '',
                textVisible: true,
                theme: 'z',
                html: ""
            });
        }else{
            $.mobile.loading('hide');
        }
    }

    /*Settings Stuff*/

});

function api(type){
    return 'https://getnomon.com/api.php?api=' + type;
}

function validateEmail(emailAddress) {
    var pattern = new RegExp(/^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))))*(((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(\x22)))@((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?$/i);
    return pattern.test(emailAddress);
};

function guid() {
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
        var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
        return v.toString(16);
    });
}
/*function doNotCall(){
    setInterval(function() {
        //call $.ajax here
    }, 5000); //5 seconds
}*/